gnosis/cowswap
# [Affiliate] Update banner closed state across tabs

# Summary

Closes #1818 #1812

This PR handles the case where the user has multiple tabs of the app and the Affiliate banner is closed in one tab. The state now is reflected across all the tabs.

There was a race condition happening as seen in [Elena's recording](https://watch.screencastify.com/v/KhjP4yhGed3YHizpGb1h) due to how the redux state is updated and also because of the redux-localstorage-simple middleware. Background: the global state in memory is persisted in local storage within each second when there's an action dispatched, if the user has multiple tabs opened, it can happen than one tab updates its global state and hence local storage after 1 second, then the user switch tabs and the other app instance had an old state in memory, which after another second updates local storage overriding the previous (and newer) state.

  # To Test

1. Paste an affiliate link into the url
2. Wait for the Affiliate banner to appear and duplicate the tab
3. Close the banner in one tab, the banner state should be reflected in the other tab
